jobsdb/jobsdb.go

```diff
@@ -873,7 +866,7 @@ func (jd *HandleT) writerSetup(ctx context.Context) {
 
 	//If no DS present, add one
 	if len(jd.datasetList) == 0 {
-		jd.addNewDS(appendToDsList, dataSetT{})
+		jd.addDS(jd.newDataSetStruct(jd.computeNewIdxForAppend()), true)
```

Comment:
Not in the scope of this PR, but I find `newDataSetStruct` method name a unusual:

```suggestion
		jd.addDS(jd.newDataSet(jd.computeNewIdxForAppend()), true)
```
or 

```suggestion
		jd.addDS(NewDataSet(jd.tablePrefix, jd.computeNewIdxForAppend()), true)
```